app/Main.hs

```diff
@@ -79,24 +81,10 @@ main = do
         putStrLn ""
         putStrLn "grep version: "
         runEffect (grepVersion >-> P.take 1 >-> P.map ("    " <>) >-> stdoutText)
-    printHelp = T.putStrLn helpText
-
-helpText :: Text
-helpText = T.unlines
-    [ "vgrep, a pager for grep"
-    , ""
-    , "Usage:"
-    , "    as a drop-in replacement for `grep -r`:"
-    , "        vgrep [GREP_OPTION...] PATTERN [FILE]"
-    , ""
-    , ""
-    , "    at the end of a pipeline:"
-    , "        ... | vgrep [-nH] PATTERN"
-    , ""
-    , "    as a pager for `grep -nH` output:"
-    , "        grep -nH [GREP_OPTION...] PATTERN [FILE] | vgrep"
-    , ""
-    , "For documentation on grep options an patterns see `grep --help`." ]
+    printHelp = putStrLn helpText
+        where helpText = $(fmap (LitE . StringL) (runIO (readFile "help.txt")))
```

Comment:
Consider [`stringE`][1]. TH is *full* of these kinds of shortcuts, usually we don’t have to use the constructors directly.

[1]: http://hackage.haskell.org/package/template-haskell-2.11.0.0/docs/src/Language.Haskell.TH.Lib.html#stringE